relations/src/preimage/tests.rs

```diff
@@ -0,0 +1,99 @@
+use ark_bls12_381::Bls12_381;
+use ark_crypto_primitives::SNARK;
+use ark_ec::bls12::Bls12;
+#[cfg(test)]
+use ark_ff::BigInteger256;
+use ark_ff::Fp256;
+use ark_groth16::{Groth16, Proof, VerifyingKey};
+#[cfg(test)]
+use ark_relations::r1cs::{ConstraintSynthesizer, ConstraintSystem};
+use ark_std::vec::Vec;
+use liminal_ark_poseidon::hash;
+
+use crate::{preimage::PreimageRelation, relation::state::FullInput, CircuitField, GetPublicInput};
+
+#[test]
+fn preimage_constraints_correctness() {
+    let preimage = CircuitField::from(17u64);
+    let image = hash::one_to_one_hash([preimage]);
+
+    let circuit: PreimageRelation<FullInput> = PreimageRelation::new(Some(preimage), Some(image));
+
+    let cs = ConstraintSystem::new_ref();
+    circuit.generate_constraints(cs.clone()).unwrap();
+
+    let is_satisfied = cs.is_satisfied().unwrap();
+    if !is_satisfied {
+        println!("{:?}", cs.which_is_unsatisfied());
+    } else {
+        println!("preimage circuit size: {:?}", cs.num_constraints());
+    }
+
+    assert!(is_satisfied);
+}
+
+#[test]
+fn unsatisfied_preimage_constraints() {
+    let true_preimage = CircuitField::from(17u64);
+    let fake_image = hash::one_to_one_hash([CircuitField::from(19u64)]);
+    let circuit: PreimageRelation<FullInput> =
+        PreimageRelation::new(Some(true_preimage), Some(fake_image));
+
+    let cs = ConstraintSystem::new_ref();
+    circuit.generate_constraints(cs.clone()).unwrap();
+
+    let is_satisfied = cs.is_satisfied().unwrap();
+
+    assert!(!is_satisfied);
+}
+
+#[test]
+pub fn preimage_proving_and_verifying() {
```

Comment:
doesn't have to be `pub`

I know that this is a low nit, but currently this module exports other functions and it is a bit confusing which things are actually used outside

Reply:
done